web/src/popover_menus.js

```diff
@@ -731,4 +776,126 @@ export function initialize() {
+                return send_at_time;
+            }
+            case "send_later_today": {
+                const send_time = send_later_today[send_later_in].time;
+                const date = new Date();
+                const send_at_time =
+                    format(date.setDate(date.getDate()), "MMM d yyyy ") + send_time;
+                return send_at_time;
+            }
+            // No default
+        }
+        blueslip.error("Not a valid time.");
+        return false;
+    }
+
+    delegate("body", {
+        ...default_popover_props,
+        target: "#send_later i",
+        onShow(instance) {
+            popovers.hide_all_except_sidebars(instance);
+
+            // Only show send later options that are possible today.
+            const date = new Date();
+            const hours = date.getHours();
+            let possible_send_later_today = {};
+            if (hours <= 8) {
+                possible_send_later_today = send_later_today;
+            } else if (hours <= 15) {
+                possible_send_later_today.today_four_pm = send_later_today.today_four_pm;
+            } else {
+                possible_send_later_today = false;
+            }
+
+            const formatted_send_later_time = get_selected_send_later_time();
+
+            instance.setContent(
+                parse_html(
+                    render_send_later_popover({
+                        possible_send_later_today,
+                        send_later_tomorrow,
+                        send_later_custom,
+                        formatted_send_later_time,
+                    }),
+                ),
+            );
+            popover_instances.send_later = instance;
+            $(instance.popper).one("click", instance.hide);
+        },
+        onMount(instance) {
+            const $popper = $(instance.popper);
+            $popper.one("click", "#send-later-custom-input", () => {
+                flatpickr.show_flatpickr(
+                    $("#send_later")[0],
+                    show_schedule_confirm_button,
+                    new Date(),
+                );
+            });
+
+            $popper.one("click", ".send_later_today, .send_later_tomorrow", (e) => {
+                const send_at_time = set_compose_box_schedule(e.currentTarget);
+                const not_from_flatpickr = true;
+                show_schedule_confirm_button(send_at_time, not_from_flatpickr);
+
+                instance.hide();
+                e.stopPropagation();
+                e.preventDefault();
+            });
+
+            $popper.one("click", "#clear_compose_schedule_state", () => {
+                selected_send_later_time = undefined;
+                $("#schedule-confirm").hide();
+                $("#compose-send-button").show();
+            });
+        },
+        onHidden(instance) {
+            instance.destroy();
+            popover_instances.send_later = undefined;
+        },
+    });
```

Comment:
We're definitely going to want to move much of this code to another module, but that'll probably be considerably easier once @shameondev makes it easier to host popovers from other modules, so maybe it's OK to let this be for now.

Reply:
It would be great if we were ready to do this right now, but it may require a lot of effort until my part is ready. So for now, I think we should leave it as it is. However, we definitely need to prioritize this work to avoid increasing our tech debt.

Reply:
Okay, sure!

Reply:
OK, added this to the follow-up TODO list.